### PR TITLE
chore: Support GMS and v3 product flavors for demo app

### LINKED ISSUE #729: Add additional product flavor to demo app
**Is your feature request related to a problem? Please describe.**
The current `demo` app is to demo features for the GMS distribution of the Utility library. This feature request is to add an additional product flavor to the `demo` app for the standalone V3 BETA distribution of the Utility library.

 **Describe the solution you'd like**
Define 2 product flavors: `gms` and `v3` so that the `demo` app can be easily configure to run either the GMS or the V3 BETA version of the library.

 **Describe alternatives you've considered**
N/A

 **Additional context**
N/A

### PR DESCRIPTION
**Pull request by barbeau**
~~**Work-in-progress - please do not merge**~~

* `gms` depends on Google Maps SDK v2 (which depends on Google Play Services)
* `v3` depends on the Google Maps SDK v3 beta standalone (which does not depend on Google Play Services)

TODO:
- [x] v3 flavor debug variant - `Cannot fit requested classes in a single dex file (# methods: 69686 > 65536)` - turn on multidex
- [x] Test the startGmsDemo and startV3Demo Gradle tasks
- [x] Update docs to reflect both product flavors
- [x] Some region tags moved - the path for the source file for the https://developers.google.com/maps/documentation/android-sdk/utility/multilayer docs needs to be changed to `demo/src/gms/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java`.

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
- [x] Will this cause breaking changes to existing Java or Kotlin integrations? If so, ensure the commit has a `BREAKING CHANGE` footer so when this change is integrated a major version update is triggered. See: https://www.conventionalcommits.org/en/v1.0.0/

Closes #729 🦕

### CODE REVIEW

**Comment by barbeau** [comment]
@arriolac This is ready for review. Note that the source files moved out of the `main` directory and the region tags in `MultiLayerDemoActivity` are used to generate the https://developers.google.com/maps/documentation/android-sdk/utility/multilayer docs. The new path for that file is `demo/src/gms/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java`. Could you update this elsewhere if it needs to be updated?

File: demo/build.gradle [hunk 1]
```diff
@@ -54,18 +70,55 @@ android {
 
 // [START maps_android_utils_install_snippet]
 dependencies {
-    implementation 'com.google.maps.android:android-maps-utils:2.2.0'
```
Comment (barbeau) [inline demo/build.gradle:57]: I just realized that this line was part of the documentation I believe on https://developers.google.com/maps/documentation/android-sdk/utility/setup.

@arriolac Thoughts on how to handle this when the dependencies are moved to the individual flavors?

**Comment by alexaguilar89** [comment]
Please chance

File: demo/build.gradle (hunk 1 above)
Comment (arriolac) [reply demo/build.gradle:57]: I'll have to update the snippet on my end once this change has been merged.

File: demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    public final static String TAG = "MultiDemo";
+
+    @Override
+    protected void startDemo(boolean isRestore) {
+        if (!isRestore) {
+            getMap().moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(51.403186, -0.126446), 10));
+        }
+
+        // Shared object managers - used to support multiple layer types on the map simultaneously
+        // [START maps_multilayer_demo_init1]
```
Comment (arriolac) [inline demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java:67]: Would you mind removing the region tags in this file? I actually migrated all snippets displayed on GMP Android docs to this repo: https://github.com/googlemaps/android-samples/tree/main/snippets so these region tags are no longer used.

**Review by arriolac — APPROVED** [review]
Just a minor comment regarding cleanup, otherwise, LGTM.

File: demo/proguard-rules.pro [hunk 3]
```diff
@@ -19,3 +19,6 @@
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
 #-renamesourcefileattribute SourceFile
+
+# Needed to avoid crash on minified app release with Maps SDK v3.1 beta (https://issuetracker.google.com/issues/148084488)
+-keep,allowoptimization class com.google.android.libraries.maps.** { *; }
```
Comment (arriolac) [inline demo/proguard-rules.pro:24]: I'll add a note in https://developers.google.com/maps/documentation/android-sdk/v3-client-migration about this issue.

File: demo/build.gradle [hunk 4]
```diff
@@ -45,6 +47,20 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
Comment (arriolac) [inline demo/build.gradle:47]: Perhaps `minifyEnabled` should be set to `true` here?

File: demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java (hunk 2 above)
Comment (barbeau) [reply demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java]: Done in https://github.com/googlemaps/android-maps-utils/pull/811/commits/05720006b0f217dffc2083bb9c844fdda3d3360b

File: demo/build.gradle (hunk 4 above)
Comment (barbeau) [reply demo/build.gradle:47]: Done in https://github.com/googlemaps/android-maps-utils/pull/811/commits/60e9881cf269b168a82319dadfe20c02d951a0ff

File: demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java (hunk 2 above)
Comment (barbeau) [reply demo/src/v3/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java]: Also in https://github.com/googlemaps/android-maps-utils/pull/811/commits/5b3824e4fb7c19061aeeb2c648ba9a11f04071bc for v3 variant

**Pull request merged**